spec/controllers/concerns/ial2_profile_concern_spec.rb

```diff
@@ -0,0 +1,213 @@
+      let!(:profile) { create(:profile, :in_person_verification_pending, user:) }
+
+      context 'when the profile can be saved to cache' do
+        before do
+          allow(cacher).to receive(:save)
+          allow(Pii::Cacher).to receive(:new).and_return(cacher)
+          subject.cache_profiles(password)
+        end
+
+        it 'stores the decrypted profile in cache' do
+          expect(cacher).to have_received(:save).with(password, profile)
+        end
+      end
+
+      context 'when the profile can not be saved to cache' do
+        before do
+          allow(cacher).to receive(:save).and_raise(encryption_error)
+          allow(Pii::Cacher).to receive(:new).and_return(cacher)
+          allow(analytics).to receive(:profile_encryption_invalid)
+          subject.cache_profiles(password)
+        end
+
+        it 'deactivates the profile with reason encryption_error' do
+          expect(profile.reload).to have_attributes(
+            active: false,
+            deactivation_reason: 'encryption_error',
+          )
+        end
+
+        it 'logs the profile_encryption_invalid analytic' do
+          expect(analytics).to have_received(:profile_encryption_invalid).with(
+            error: encryption_error.message,
+          )
+        end
+      end
+    end
+
+    context 'when the user has an active profile' do
+      let!(:profile) { create(:profile, :active, user:) }
+
+      context 'when the profile can be saved to cache' do
+        before do
+          allow(cacher).to receive(:save)
+          allow(Pii::Cacher).to receive(:new).and_return(cacher)
+          subject.cache_profiles(password)
+        end
+
+        it 'stores the decrypted profile in cache' do
+          expect(cacher).to have_received(:save).with(password, profile)
+        end
+      end
+
+      context 'when the profile can not be saved to cache' do
+        before do
+          allow(cacher).to receive(:save).and_raise(encryption_error)
+          allow(Pii::Cacher).to receive(:new).and_return(cacher)
+          allow(analytics).to receive(:profile_encryption_invalid)
+          subject.cache_profiles(password)
+        end
+
+        it 'deactivates the profile with reason encryption_error' do
+          expect(profile.reload).to have_attributes(
+            active: false,
+            deactivation_reason: 'encryption_error',
+          )
+        end
+
+        it 'logs the profile_encryption_invalid analytic' do
+          expect(analytics).to have_received(:profile_encryption_invalid).with(
+            error: encryption_error.message,
+          )
+        end
+      end
+    end
+
+    context 'when the user has both an active profile and pending profile' do
+      let(:pending_profile) { create(:profile, :in_person_verification_pending, user:) }
+      let(:active_profile) { create(:profile, :active, user:) }
+
+      context 'when the active profile was activated before the pending profile as created' do
```

Comment:
```suggestion
      context 'when the active profile was activated before the pending profile was created' do
```

Is that how this is supposed to read?

Reply:
Nice catch! I will update that!